src/SDL_shadercross.c

```diff
@@ -2098,13 +2284,18 @@ static void *SDL_ShaderCross_INTERNAL_CompileFromSPIRV(
         if (createInfo.props != 0) {
             SDL_DestroyProperties(createInfo.props);
         }
+        SDL_free(pipelineInfo);
     } else {
         SDL_GPUShaderCreateInfo createInfo;
-        SDL_ShaderCross_GraphicsShaderMetadata *shaderInfo = (SDL_ShaderCross_GraphicsShaderMetadata *)metadata;
-        SDL_ShaderCross_ReflectGraphicsSPIRV(
-            info->bytecode,
-            info->bytecode_size,
-            shaderInfo);
+        SDL_ShaderCross_GraphicsShaderMetadata *shaderInfo =
+            SDL_ShaderCross_ReflectGraphicsSPIRV(
+                info->bytecode,
+                info->bytecode_size,
+                metadataProps);
+
+        if (shaderInfo == NULL) {
+
+        }
```

Comment:
`SDL_ShaderCross_ReflectGraphicsSPIRV` should have already done `SDL_SetError`.
```suggestion
        if (shaderInfo == NULL) {
            SDL_ShaderCross_INTERNAL_DestroyTranspileContext(transpileContext);
            return NULL;
        }
```